### ISSUE TITLE
JUnit 5 depletes heap for Maven tests

### ISSUE THREAD

**Issue by huffstler** [junit-team/junit5#1346]
## Overview

For some reason or another, trying to run a suite of tests for a project depletes the heap in the jvm, but only when running the suite, and only when all the tests are there. For the sake of keeping the issue clean, I've created gists with code and output in them and will link where needed.

[This](https://gist.github.com/huffstler/8a6b441703e816c3e7e2df93bb1a2aa3#file-etlprops-java) is one of two test classes I have. Running the test _suite_ outputs this error currently: 

``` bash
mvn -X deploy <enter>
```
[output](https://gist.github.com/huffstler/8a6b441703e816c3e7e2df93bb1a2aa3#file-mvn-x-deploy)
(Picture of ram usage to go with output)
![ram_usage](https://user-images.githubusercontent.com/9092204/37988713-a01cec0c-31cf-11e8-916b-877d1190c5aa.PNG)

_However_, running just `ETLPropsTest.java` produces [this output](https://gist.github.com/huffstler/8a6b441703e816c3e7e2df93bb1a2aa3#file-running_only_etlpropstest) instead. 
TL;DR: The functions and fields need to be static.

Now, commenting out the blank tests in the first file I linked (`ETLPropsTests.java`) and leaving only the code that actually does something lets the entire suite pass. 
[Like so](https://gist.github.com/huffstler/8a6b441703e816c3e7e2df93bb1a2aa3#file-commentedout-java). 

I would have expected that removing tests that don't do anything wouldn't affect the memory usage when testing, but that doesn't seem to be the case.

I honestly can't tell currently if this is a Maven or JUnit issue right now (tbh, it's probably user error) so I'll open up an issue with Apache Maven and link it over for consistency.

**Comment by geo-m** [comment]
Just a curious user here, but this also happen if you set Lifecycle.PER_CLASS ?

**Comment by marcphilipp** [comment]
Our Surefire provider is not compatible with Surefire 2.20. Please try using Surefire 2.19.1.

**Comment by sormuras** [comment]
@huffstler As Surefire 2.22.0 is released, may you please re-run your setup and verify if the issue still persists using the latest-and-greatest version with native JUnit Platform support?

Thanks in advance.

**Comment by sormuras** [comment]
Closing this issue for now -- please report new issues at https://maven.apache.org/surefire/issue-tracking.html

@huffstler We're interested in your feedback regarding your issue when using Surefire 2.22.0.
